src/components/Elements/TOC/TOC.tsx

```diff
@@ -0,0 +1,74 @@
+import type { MarkdownHeading } from "astro"
+import { onMount, type Component } from "solid-js"
+
+export type TOCProps = {
+  headings: MarkdownHeading[]
+}
+
+export const TOC: Component<TOCProps> = (props) => {
+  let observer: IntersectionObserver
+  let activeHeadingCount: number = 0
+  let pendingDeactivateHeadingElements: HTMLElement[] = []
+
+  const headings = props.headings
+  const headingElements = headings
+    .filter((heading) => heading.depth === 2 || heading.depth === 3)
+    .map((heading) => (
+      <li class="mb-2 ms-4">
+        <div
+          class={`absolute ${
+            heading.depth === 2 ? "-start-[6px] mt-1.5 size-3" : "-start-[3px] mt-2 size-1.5"
+          } rounded-full border border-border-muted bg-border-muted`}
```

Comment:
Nitpick: 可読性と安全性のために、テンプレート文字列よりもtailwind-mergeを用いた方が良いと思われます。また、より複雑な場合な条件分岐を含む場合にはclsxを使った方が良いかもしれません

```suggestion
          class={twMerge(
            "absolute rounded-full border border-border-muted bg-border-muted",
            heading.depth === 2 ? "-start-[6px] mt-1.5 size-3" : "-start-[3px] mt-2 size-1.5",
          )}
```